tuf/client/updater.py

```diff
@@ -1592,20 +1592,22 @@ def _update_metadata_if_changed(self, metadata_role,
     # should always be 'snapshot'.  'snapshot.json' specifies all roles
     # provided by a repository, including their version numbers.
     if metadata_role == 'snapshot' or metadata_role.startswith('targets'):
-      gzip_metadata_filename = uncompressed_metadata_filename + '.gz'
       if 'gzip' in self.metadata['current']['root']['compression_algorithms']:
         compression = 'gzip'
-        
-        logger.debug('Compressed version of ' + \
-                repr(uncompressed_metadata_filename) + ' is available at ' + \
-                repr(gzip_metadata_filename) + '.')
+        gzip_metadata_filename = uncompressed_metadata_filename + '.gz'
+        logger.debug('Compressed version of ' +
+          repr(uncompressed_metadata_filename) + ' is available at ' +
+          repr(gzip_metadata_filename) + '.')
+      
       else:
-        logger.debug('Compressed version of ' + \
-                     repr(uncompressed_metadata_filename) + ' not available.')
-
-    # Simply return if the file has not changed, according to the metadata
-    # about the uncompressed file provided by the referenced metadata.
-    if not self._versioninfo_has_changed(uncompressed_metadata_filename,
+        logger.debug('Compressed version of ' +
+          repr(uncompressed_metadata_filename) + ' not available.')
+
+    # Simply return if the metadata for 'metadata_role' has not been updated,
```

Comment:
I don't understand why we're doing this check after the compression check instead of before. I think it should be the other way around. It confused me for a bit. (TBC: Block 1595-1604 switch with 1606-1617.)